src/Symfony/Component/HttpFoundation/EventStreamResponse.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * Represents a streaming HTTP response for sending server events
+ * as part of the Server-Sent Events (SSE) streaming technique.
+ *
+ * @see ServerEvent
+ *
+ * @author Yonel Ceruto <open@yceruto.dev>
+ *
+ * Example usage:
+ *
+ *     return new EventStreamResponse(function () {
+ *         while (true) {
+ *             yield new ServerEvent(time(), type: 'ping');
+ *
+ *             if (connection_aborted()) {
+ *                 break;
+ *             }
+ *
+ *             sleep(1);
+ *         }
+ *     });
+ */
+class EventStreamResponse extends StreamedResponse
+{
+    /**
+     * @param int $retry The event reconnection time in milliseconds
+     */
+    public function __construct(?callable $callback = null, int $status = 200, array $headers = [], private int $retry = 0)
+    {
+        $headers += [
+            'Content-Type' => 'text/event-stream',
+            'Cache-Control' => 'no-cache',
+            'Connection' => 'keep-alive',
```

Comment:
We should also set these legacy cache headers and the `X-Accel-Buffering` NGINX header by default to prevent issues with many reverse proxies (like NGINX Ingress for Kubernetes, which is very popular): https://github.com/dunglas/mercure/blob/main/subscribe.go#L218

Reply:
Headers updated!